ieeeuoft/newhacks-site-2024
# Add Qr code scanner code

## Overview

- Resolves #<number>
- add QR code scanner support
- admins can scan QR codes, and participants that have accepted and rsvped can view thier QR codes


## Unit Tests Created

- n/a


## Steps to QA

- run migrations with `python manage.py migrate`
- create an admin user and log in to localhost:8000 --> you should be able to view admin dashboard
   - click on qr code scanning, this should take you to qr scanning page
   - currently there is no active event, in order for scanning to be active you need to update the `EVENT_START_DATE = datetime(2024, 10, 26, 8, 0, 0, tzinfo=TZ_INFO)` to earlier so you can test it and adjust one of these sign in event start dates to your current time so its active following in `settings/__init__.py`
   ```
SIGN_IN_TIMES = [
    {
        "name": "sign_in",
        "description": "Hackathon Sign In",
        "time": datetime(2023, 11, 4, 10, 0, 0, tzinfo=TZ_INFO),  # Nov 4th @ 9am
    },
    {
        "name": "lunch1",
        "description": "Lunch Day 1",
        "time": datetime(2023, 11, 4, 12, 30, 0, tzinfo=TZ_INFO),  # Nov 4th @ 12:30pm
    },
    {
        "name": "dinner1",
        "description": "Dinner Day 1",
        "time": datetime(2023, 11, 4, 18, 0, 0, tzinfo=TZ_INFO),  # Nov 4th @ 6pm
    },
    {
        "name": "breakfast2",
        "description": "Breakfast Day 2",
        "time": datetime(2023, 11, 5, 8, 0, 0, tzinfo=TZ_INFO),  # Nov 5th @ 8am
    },
    {
        "name": "lunch2",
        "description": "Lunch Day 2",
        "time": datetime(2023, 11, 5, 11, 45, 0, tzinfo=TZ_INFO),  # Nov 5th @ 11:45pm
    },
]```

- In order to test scanning a QR code you need a QR code --> first create a user with an application who has been accepted and has rsvp(ask me if you don't know how to do this), sign in to dashboard and it should have QR code
- then just go to admin site and show off this QR code(i made a screenshot of the QR code and then showed to the scanner)
- it should work and give details of your dietary requirements and t shirt size
